src/dd/GateMatrixDefinitions.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2024 Chair for Design Automation, TUM
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+#include "dd/GateMatrixDefinitions.hpp"
+
+#include "dd/DDDefinitions.hpp"
+#include "ir/operations/OpType.hpp"
+
+#include <complex>
+#include <unordered_map>
+#include <vector>
+
+namespace dd {
+
+const std::unordered_map<qc::OpType,
+                         GateMatrix (*const)(const std::vector<fp>&)>
+    MATS_GENERATORS{{
+        // clang-format off
```

Comment:
Given how you purposely disable formatting here, it might be worth to actually format the matrices as matrices instead of linear lists 🤷🏼